src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelNotificationBuilderTest.java

```diff
@@ -200,6 +248,116 @@ public void testNoActionButtons() {
+        verify(builderSpy).setChannelId("12345");
+        assertEquals(notification.getChannelId(), "12345");
+    }
+
+    public void testNoChannelId() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy).setChannelId(anyString());
+        assertEquals(notification.getChannelId(), MixpanelPushNotification.NotificationData.DEFAULT_CHANNEL_ID);
+    }
+
+    public void testSubText() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_subtxt", "SUBTEXT");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy).setSubText("SUBTEXT");
+    }
+
+    public void testNoSubText() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy, never()).setSubText("MESSAGE");
+    }
+
+    public void testTicker() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_ticker", "TICK");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy).setTicker("TICK");
+    }
+
+    public void testNoTicker() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy).setTicker("MESSAGE");
+    }
+
+    public void testSticky() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_sticky", "true");
+        Notification notification = mpPushSpy.createNotification(intent);
+        int flag = notification.flags;
+        assertTrue((flag | Notification.FLAG_AUTO_CANCEL) != flag);
+    }
+
+    public void testNoSticky() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        Notification notification = mpPushSpy.createNotification(intent);
+        int flag = notification.flags;
+        assertEquals(flag | Notification.FLAG_AUTO_CANCEL, flag);
+    }
+
+    public void testTimestamp() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_time", "2014-10-02T15:01:23.045123456Z");
+        Notification notification = mpPushSpy.createNotification(intent);
+        Instant instant = Instant.parse( "2011-05-03T11:58:01Z" );
+        verify(builderSpy).setWhen(instant.toEpochMilli());
+    }
+
+    public void testNoTimestamp() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        verify(builderSpy, never()).setWhen(any(Long.class));
```

Comment:
Just curious: do we need to `setWhen` to the current time or does it just default to it?

Reply:
I believe it the default is handled when the notification arrives to the device. I did also need to add setShowWhen(bool) as well though.